### DISCUSSION TITLE
[golang] Folder structured not honored

### DISCUSSION THREAD

**Discussion by TaiSHiNet** [bufbuild/buf#3076] [Q&A] [answered]
Hi,

I'm trying to replicate the behavior of:
```
protoc -I protobuf-schema --go_out=pkg --go_opt=module=github.com/OurCompany/go-protobuf/pkg  protobuf-schema/*.proto
```

Which stores our `*.pb.go` files honoring the module path such as: `pkg/health/health.pb.go`
The behavior I'm seeing is that files get generated like `pkg/health.pb.go` instead.

Versions:
```
$ protoc --version
libprotoc 26.1
$ protoc-gen-go --version
protoc-gen-go v1.34.2
$ buf --version
1.32.2
```

``` health.proto
syntax = "proto3";

package health.v1;
// option go_package = "github.com/OurCompany/go-protobuf/pkg/health";
// tested without being commented

message HealthCheckRequest {
  string service = 1;
}

message HealthCheckResponse {
  enum ServingStatus {
    UNKNOWN = 0;
    SERVING = 1;
    NOT_SERVING = 2;
    SERVICE_UNKNOWN = 3;  // Used only by the Watch method.
  }
  ServingStatus status = 1;
}

service Health {
  rpc Check(HealthCheckRequest) returns (HealthCheckResponse);

  rpc Watch(HealthCheckRequest) returns (stream HealthCheckResponse);
}
```

``` buf.gen.yaml
version: v2
managed:
  enabled: true
  disable:
    - file_option: go_package
      module: buf.build/googleapis/googleapis
    - file_option: go_package_prefix
      module: buf.build/grpc-ecosystem/grpc-gateway
  override:
    - file_option: go_package
      value: github.com/OurCompany/go-protobuf/pkg
plugins:
  - remote: buf.build/protocolbuffers/go
    out: pkg
    opt:
      - module=github.com/OurCompany/go-protobuf/pkg
  - remote: buf.build/grpc/go
    out: pkg
    opt:
      - module=github.com/OurCompany/go-protobuf/pkg
  - remote: buf.build/grpc-ecosystem/gateway
    out: pkg
    opt:
      - module=github.com/OurCompany/go-protobuf/pkg

```

^ tried with `paths=import` and `paths=source_relative`

Any pointer is greatly appreciated

**Comment by TaiSHiNet** [comment]
After messing with the linter and further reading documentation I saw that the folder structure is inherited from the `*.proto` source.
Is there a way to override this?

**Reply by bufdev** [reply, marked as answer]
Thanks for your question here. There isn't a way to override this - `buf` encourages standardized `.proto` file layouts, and having a 1-1 mapping between your directory structure and package definitions is really almost a requirement nowadays in Protobuf world, even outside of `buf`. If you can provide us a reproducible setup in i.e. a temporary github repository, we're happy to help advise you on how to proceed!

**Reply by TaiSHiNet** [reply]
Hi! I've decided to follow guidelines and adapt an existing protobuf repo to conform to standards.
It's a bit more work but it pays off!

Appreciate the response
